File: BaekJoon/DP/11053.py
import sys


N = int(input())

li = list(map(int, sys.stdin.readline().split()))

dp = [1]*N

for i in range(N):
    for j in range(i):
        if li[j] < li[i]:
            dp[i] = max(dp[i],dp[j]+1)

print(max(dp))

File: BaekJoonSwift/BaekJoonSwift/DP (2)/11053.swift
// 11053 가장 긴 증가하는 부분 수열

let n = Int(readLine()!)!
let arr = readLine()!.split(separator: " ").map { Int(String($0))! }

var dp = Array(repeating: 1, count: n)

for i in 1..<n {
    for j in 0..<i {
        if arr[i] > arr[j] {
            dp[i] = max(dp[i], dp[j]+1)
        }
    }
}

print(dp.max()!)
